docs: clarify CommonJS with `optimizeDeps.exclude` (#3961)
vitejs/vite: docs/config/index.md

  Dependencies to exclude from pre-bundling.

  :::warning CommonJS

>   CommonJS dependencies cannot be excluded from optimization. If an ESM dependency has a CommonJS dependency, it also cannot be excluded.

Reviewer: ```suggestion
  CommonJS dependencies should not be excluded from optimization. If an ESM dependency has a nested CommonJS dependency, it should not be excluded as well.
```

Reviewer: I am thinking if we should document to force include the nested dep, eg:

```js
{
  optimizeDeps: {
    exclude: ['esm-dep'],
    include: ['esm-dep-nesed-cjs']
  }
}
```

Author: Does that really work? If so, let's document it

Reviewer 2: Yes, it does. I see people doing it all the time for SvelteKit. But it's a bit annoying that it needs to be done at all in my opinion. If I exclude a library I generally only want to exclude that specific library and not it's dependencies

Reviewer 3: Using `optimizeDeps.include` on a transitive dependency is fragile and is dependent on the flat `node_modules` used by npm. When `pnpm`'s default nested structure is used, for example, this breaks because the transitive dependency isn't found:

```shell
> Failed to resolve force included dependency: clipboard-copy
Error: Failed to resolve force included dependency: clipboard-copy
    at optimizeDeps
```

Adding the transitive dependency as a direct dependency appears to avoid this issue, but that seems like it'd cause problems in other ways.

Reviewer 2: Shouldn't all transitive dependencies be optimized by default? Perhaps the issue here is that Vite's dependency crawling mechanism doesn't work with `.svelte` files?

Reviewer: I'll merge this for now. Please create a new thread for the future discussion.
